library/Icingadb/Widget/ItemList/RedundancyGroupListItem.php

```diff
@@ -0,0 +1,153 @@
+            'span',
+            Attributes::create(['class' => 'subject']),
+            Text::create($this->item->display_name)
+        );
+    }
+
+    protected function assembleVisual(BaseHtmlElement $visual): void
+    {
+        $stateText = $this->state->failed ? 'problem' : 'ok';
+        $stateBall = new StateBall($stateText, $this->getStateBallSize());
+
+        $visual->addHtml($stateBall);
+    }
+
+    protected function assembleCaption(BaseHtmlElement $caption): void
+    {
+        $filter = Filter::equal('id', $this->item->id);
+        $relations = [
+            'from',
+            'from.to.host',
+            'from.to.host.state',
+            'from.to.service',
+            'from.to.service.state'
+        ];
+
+        $summary = RedundancyGroupParentStateSummary::on($this->getDb())
+            ->with($relations)
+            ->filter($filter);
+
+        $members = RedundancyGroup::on($this->getDb())
+            ->columns([
+                'id' => 'id',
+                'parent_output' => new Expression(
+                    'CASE WHEN redundancy_group_from_to_host_state.output IS NULL'
+                    . ' THEN redundancy_group_from_to_service_state.output'
+                    . ' ELSE redundancy_group_from_to_host_state.output END'
+                ),
+                'parent_long_output' => new Expression(
+                    'CASE WHEN redundancy_group_from_to_host_state.long_output IS NULL'
+                    . ' THEN redundancy_group_from_to_service_state.long_output'
+                    . ' ELSE redundancy_group_from_to_host_state.long_output END'
+                ),
+                'parent_checkcommand_name' => new Expression(
+                    'CASE WHEN redundancy_group_from_to_host.checkcommand_name IS NULL'
+                    . ' THEN redundancy_group_from_to_service.checkcommand_name'
+                    . ' ELSE redundancy_group_from_to_host.checkcommand_name END'
+                ),
+                'parent_last_state_change' => new Expression(
+                    'CASE WHEN redundancy_group_from_to_host_state.last_state_change IS NULL'
+                    . ' THEN redundancy_group_from_to_service_state.last_state_change'
+                    . ' ELSE redundancy_group_from_to_host_state.last_state_change END'
+                ),
+                'parent_severity' => new Expression(
+                    'CASE WHEN redundancy_group_from_to_host_state.severity IS NULL'
+                    . ' THEN redundancy_group_from_to_service_state.severity'
+                    . ' ELSE redundancy_group_from_to_host_state.severity END'
+                )
+            ])
+            ->with($relations)
+            ->filter($filter)
+            ->orderBy([
+                'parent_severity',
+                'parent_last_state_change',
+            ], 'DESC');
+
+        $this->applyRestrictions($members);
+
+        /** @var RedundancyGroup $data */
+        $data = $members->first();
```

Comment:
I don't quite like that this is evaluated per group result.

Keep it as it is for now, but update it so that it matches the changes made to the statistics and badges implementation.

I have an alternative implementation in mind, but that's now far to complex for this change and so will be implemented separately. Prior approval of this change, I will request to remove this to not show any badges or output for groups. Until then, it's kept to test the statistics and badges implementation changes.